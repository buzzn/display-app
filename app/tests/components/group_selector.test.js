import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GroupSelector from '../../components/group_selector';

describe('group selector component tests', () => {
  test('renders group list correctly', () => {
    const groups = [
      { id: '1', name: 'group1' },
      { id: '2', name: 'group2' },
    ];
    const tree = renderer.create(
      <GroupSelector groups={ groups }/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls callback with proper group id', () => {
    const onGroupSelect = jest.fn();
    const groups = [
      { id: '1', name: 'group1' },
      { id: '2', name: 'group2' },
    ];
    const rendered = shallow(
      <GroupSelector groups={ groups } onGroupSelect={ onGroupSelect }/>
    );
    rendered.find('select').simulate('change', { target: { value: groups[1].id } });
    // rendered.find('select').simulate('select', groups[1].id);
    expect(onGroupSelect.mock.calls[0][0]).toBe(groups[1].id);
  });
});
