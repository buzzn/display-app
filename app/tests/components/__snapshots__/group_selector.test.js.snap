// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group selector component tests renders group list correctly 1`] = `
<select
  onChange={[Function]}
>
  <option
    value=""
  >
    -----
  </option>
  <option
    value="1"
  >
    group1
  </option>
  <option
    value="2"
  >
    group2
  </option>
</select>
`;
